browser/resources/settings/getting_started_page/brave_account_row.html.ts

```diff
@@ -3,37 +3,90 @@
       </div>
       <div class="title-and-description">
-        <div class="title">
-          ${this.signedIn
-            ? 'John Doe'
-            : loadTimeData.getString('braveAccountRowTitle')}
-        </div>
-        <div class="description">
-          ${this.signedIn
-            ? 'johndoe@gmail.com'
-            : loadTimeData.getString('braveAccountRowDescription')}
-        </div>
+        <div class="title">${title}</div>
+        ${descriptions.map(
+          desc => html`<div class="description">${desc}</div>`)}
       </div>
-      <leo-button kind=${this.signedIn ? 'outline' : 'filled'}
-                  size="small"
-                  @click=${this.onButtonClicked}>
-        ${this.signedIn
-          ? loadTimeData.getString('braveAccountManageAccountButtonLabel')
-          : loadTimeData.getString('braveAccountGetStartedButtonLabel')
-        }
-      </leo-button>
+      ${button || nothing}
+    </div>
+  `
+}
+
+export function getHtml(this: SettingsBraveAccountRow) {
+  const stateHtml: Record<
+    AccountState,
+    () => ReturnType<typeof html>
+  > = {
+    [AccountState.kLoggedIn]: () => createFirstRow(
+      'John Doe',
+      'johndoe@gmail.com',
+      html`
+        <leo-button kind="outline"
+                    size="small"
+                    @click=${this.onLogOutButtonClicked}>
+          ${loadTimeData.getString('braveAccountLogOutButtonLabel')}
+        </leo-button>
+      `
+    ),
+    [AccountState.kVerification]: () => html`
+      ${createFirstRow(
+        loadTimeData.getString('braveAccountVerificationRowTitle'),
+        [
+          loadTimeData.getString('braveAccountVerificationRowDescription1'),
+          loadTimeData.getString('braveAccountVerificationRowDescription2')
+        ]
+      )}
+      <div class="second-row">
+        <leo-button kind="outline"
+                    size="small"
+                    @click=${this.onResendConfirmationEmailButtonClicked}>
+          ${loadTimeData.getString(
+            'braveAccountResendConfirmationEmailButtonLabel')}
+        </leo-button>
+        <leo-button kind="plain-faint"
+                    size="small"
+                    class="cancel-registration-button"
+                    @click=${this.onCancelRegistrationButtonClicked}>
+          ${loadTimeData.getString('braveAccountCancelRegistrationButtonLabel')}
+        </leo-button>
+      </div>
+    `,
+    [AccountState.kLoggedOut]: () => createFirstRow(
+      loadTimeData.getString('braveAccountLoggedOutRowTitle'),
+      loadTimeData.getString('braveAccountLoggedOutRowDescription'),
+      html`
+        <leo-button kind="filled"
+                    size="small"
+                    @click=${this.onGetStartedButtonClicked}>
+          ${loadTimeData.getString('braveAccountGetStartedButtonLabel')}
+        </leo-button>
+      `
+    ),
+  }
+
+  return html`
+    <div class="row-container">
+      ${this.state === undefined ? nothing : stateHtml[this.state]()}
```

Comment:
mega optional (and only if you want) but this could also be a piece of derived state if you prefer:

```ts
get loading() {
    return this.state === undefined
}
```